app/Http/Controllers/Three/ReactionController.php

```diff
@@ -67,4 +67,21 @@ public function store(Request $request, Post $post)
 
         return $this->item($reaction, $code, $meta);
     }
+
+    /**
+     * Returns all reactions for a post.
+     * GET /post/:post_id/reactions
+     *
+     * @param \Illuminate\Http\Request $request
+     * @param \Rogue\Models\Post $post
+     * @return Illuminate\Http\Response
+     */
+    public function index(Request $request, Post $post)
+    {
+        $query = Reaction::withTrashed()->where(['post_id' => $post->id]);
+
+        $reactions = Reaction::withTrashed()->where(['post_id' => $post->id])->get();
```

Comment:
hmm looks like you are doing this query twice?

Reply:
ah sorry for not catching that!